### ISSUE TITLE
Error: ParserError: Source file requires different compiler version 

### ISSUE THREAD

**Issue by mcwebdev** [Nikhil22/angular4-truffle-starter-dapp#16]
I am getting an error when trying to run the truffle commands.

`ParserError: Source file requires different compiler version (current compiler is 0.5.12+commit.7709ece9.Emscripten.clang - note that nightly builds are considered to be strictly less than the released version
pragma solidity ^0.4.2;`

`Error: Truffle is currently using solc 0.5.12, but one or more of your contracts specify "pragma solidity ^0.4.2".
Please update your truffle config or pragma statement(s).
(See https://truffleframework.com/docs/truffle/reference/configuration#compiler-configuration for information on
configuring Truffle to use a specific solc compiler version.)

Compilation failed. See above.`

**Comment by mcwebdev** [comment]
If I change the version number at the top of the `.sol` files I get a different error.

```> Compilation warnings encountered:

    /C/workspace/angular-truffle-starter-dapp/contracts/MetaCoin.sol:15:2: Warning: This declaration shadows an existing declaration.
        function MetaCoin() {
 ^ (Relevant source part starts here and spans across multiple lines).
/C/workspace/angular-truffle-starter-dapp/contracts/MetaCoin.sol:10:1: The shadowed declaration is here:
contract MetaCoin {
^ (Relevant source part starts here and spans across multiple lines).
,/C/workspace/angular-truffle-starter-dapp/contracts/Migrations.sol:11:3: Warning: This declaration shadows an existing declaration.
  function Migrations() {
  ^ (Relevant source part starts here and spans across multiple lines).
/C/workspace/angular-truffle-starter-dapp/contracts/Migrations.sol:3:1: The shadowed declaration is here:
contract Migrations {
^ (Relevant source part starts here and spans across multiple lines).

/C/workspace/angular-truffle-starter-dapp/contracts/ConvertLib.sol:4:2: SyntaxError: No visibility specified. Did you intend to add "public"?
        function convert(uint amount,uint conversionRate) returns (uint convertedAmount)
 ^ (Relevant source part starts here and spans across multiple lines).
,/C/workspace/angular-truffle-starter-dapp/contracts/MetaCoin.sol:15:2: SyntaxError: Functions are not allowed to have the same name as the contract. If you intend this to be a constructor, use "constructor(...) { ... }" to define it.
        function MetaCoin() {
 ^ (Relevant source part starts here and spans across multiple lines).
,/C/workspace/angular-truffle-starter-dapp/contracts/MetaCoin.sol:15:2: SyntaxError: No visibility specified. Did you intend to add "public"?
        function MetaCoin() {
 ^ (Relevant source part starts here and spans across multiple lines).
,/C/workspace/angular-truffle-starter-dapp/contracts/MetaCoin.sol:19:2: SyntaxError: No visibility specified. Did you intend to add "public"?
        function sendCoin(address receiver, uint amount) returns(bool sufficient) {
 ^ (Relevant source part starts here and spans across multiple lines).
,/C/workspace/angular-truffle-starter-dapp/contracts/MetaCoin.sol:27:2: SyntaxError: No visibility specified. Did you intend to add "public"?
        function getBalanceInEth(address addr) returns(uint){
 ^ (Relevant source part starts here and spans across multiple lines).
,/C/workspace/angular-truffle-starter-dapp/contracts/MetaCoin.sol:31:2: SyntaxError: No visibility specified. Did you intend to add "public"?
        function getBalance(address addr) returns(uint) {
 ^ (Relevant source part starts here and spans across multiple lines).
,/C/workspace/angular-truffle-starter-dapp/contracts/Migrations.sol:11:3: SyntaxError: Functions are not allowed to have the same name as the contract. If you intend this to be a constructor, use "constructor(...) { ... }" to define it.
  function Migrations() {
  ^ (Relevant source part starts here and spans across multiple lines).
,/C/workspace/angular-truffle-starter-dapp/contracts/Migrations.sol:11:3: SyntaxError: No visibility specified. Did you intend to add "public"?
  function Migrations() {
  ^ (Relevant source part starts here and spans across multiple lines).
,/C/workspace/angular-truffle-starter-dapp/test/TestMetacoin.sol:9:3: SyntaxError: No visibility specified. Did you intend to add "public"?
  function testInitialBalanceUsingDeployedContract() {
  ^ (Relevant source part starts here and spans across multiple lines).
,/C/workspace/angular-truffle-starter-dapp/test/TestMetacoin.sol:17:3: SyntaxError: No visibility specified. Did you intend to add "public"?
  function testInitialBalanceWithNewMetaCoin() {
  ^ (Relevant source part starts here and spans across multiple lines).
,/C/workspace/angular-truffle-starter-dapp/contracts/Migrations.sol:20:5: TypeError: Name has to refer to a struct, enum or contract.
    Migrations upgraded = Migrations(new_address);
    ^--------^

Compilation failed. See above.
Truffle v5.1.8 (core: 5.1.8)
Node v12.14.0```

**Comment by mcwebdev** [comment]
Fix : `pragma solidity >=0.4.0 <0.6.0;`

**Comment by mcwebdev** [comment]
fixed
